jaeger-core/src/main/java/com/uber/jaeger/senders/HttpSender.java

```diff
@@ -24,61 +24,81 @@
 import com.uber.jaeger.thriftjava.Process;
+import com.uber.jaeger.thriftjava.Span;
 import java.io.IOException;
-import java.net.URI;
-import java.net.URISyntaxException;
 import java.util.List;
-import org.apache.http.HttpResponse;
-import org.apache.http.HttpStatus;
-import org.apache.http.client.HttpClient;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.entity.ByteArrayEntity;
-import org.apache.http.impl.client.DefaultHttpClient;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
 import org.apache.thrift.TException;
 import org.apache.thrift.TSerializer;
-import org.apache.thrift.protocol.TBinaryProtocol.Factory;
+import org.apache.thrift.protocol.TBinaryProtocol;
 import org.apache.thrift.protocol.TProtocolFactory;
 
 public class HttpSender extends ThriftSender {
+
   private static final String HTTP_COLLECTOR_JAEGER_THRIFT_FORMAT_PARAM = "format=jaeger.thrift";
-  private static final TProtocolFactory PROTOCOL_FACTORY = new Factory();
+  private static final TProtocolFactory PROTOCOL_FACTORY = new TBinaryProtocol.Factory();
   private static final TSerializer SERIALIZER = new TSerializer(PROTOCOL_FACTORY);
+  private static final int ONE_MB_IN_BYTES = 1048576;
+  private static final MediaType MEDIA_TYPE_THRIFT = MediaType.parse("application/x-thrift");
+  private final OkHttpClient httpClient;
+  private final HttpUrl collectorUrl;
 
-  private final HttpClient httpClient = new DefaultHttpClient();
-  private final URI collectorUri;
+  /**
+   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g
+   * http://localhost:14268/api/traces
+   *
+   * Uses the default {@link okhttp3.OkHttpClient} which uses {@link okhttp3.ConnectionPool#ConnectionPool()}.
+   * Use {@link HttpSender#HttpSender(java.lang.String, int, okhttp3.OkHttpClient)} to adjust parameters.
+   */
+  public HttpSender(String endpoint) {
+    this(endpoint, ONE_MB_IN_BYTES, new OkHttpClient());
+  }
 
   /**
-   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g http://localhost:14268/api/traces
-   * @param maxPacketSize max packet size
+   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g
+   * http://localhost:14268/api/traces
+   * @param maxPayloadBytes max bytes to serialize as payload
+   * @param client a client used to make http requests
    */
-  public HttpSender(String endpoint, int maxPacketSize) {
-    super(PROTOCOL_FACTORY, maxPacketSize);
-    this.collectorUri = constructCollectorUri(endpoint);
+  public HttpSender(String endpoint, int maxPayloadBytes, OkHttpClient client) {
+    super(PROTOCOL_FACTORY, maxPayloadBytes);
+    this.collectorUrl = HttpUrl.parse(String.format("%s?%s", endpoint, HTTP_COLLECTOR_JAEGER_THRIFT_FORMAT_PARAM));
+    if (this.collectorUrl == null) {
+      throw new IllegalArgumentException("Could not parse url.");
```

Comment:
`IllegalStateException` would make sense if an object has been created, but suddenly had bad data in its internal data structure. You are here on the constructor, so, there's no "state" yet ;-)